invenio_records_resources/services/records/service.py

```diff
@@ -126,7 +125,8 @@ def search_request(self, identity, params, record_cls, preference=True):
     #
     # High-level API
     #
-    def search(self, identity, params=None, links_config=None, **kwargs):
+    def search(self, identity, params=None, links_config=None,
+               es_preference=None, **kwargs):
```

Comment:
Shall we call it `preference`?

Reply:
At that interface level (where kwargs can be anything) I preferred keeping the name less generic than "preference" so less chance of future conflict and clearer semantics (it concerns elasticsearch). At that level, knowledge of the underlying search engine is a bit exceptional.